src/libcore/iter/range.rs

```diff
@@ -325,6 +325,9 @@ impl<A: Step> Iterator for ops::RangeFrom<A> {
 #[unstable(feature = "fused", issue = "35602")]
 impl<A: Step> FusedIterator for ops::RangeFrom<A> {}
 
+#[unstable(feature = "trusted_len", issue = "37572")]
+unsafe impl<A: Step> TrustedLen for ops::RangeFrom<A> {}
+
```

Comment:
Uh oh, this gets into the discussion about the "length of a x.. range" again.. I suppose it fits, it can panic before reaching the trusted length just as much as `any_iterator.map(f)` can.

Reply:
If it panics, it diverges, which should be fine. Worst case is that e.g. a Vector is currently collecting the elements. This will leave it with uninitialized values, but it should be dropped anyway / it shouldn't be accessable anymore due to the panic.

Reply:
Linking previous discussion about why `RangeFrom` needs to have an infinite size hint, which was also about `Take<RangeFrom<_>>`: https://github.com/rust-lang/rust/pull/42315#issuecomment-305014630

Reply:
@oberien Yes, checking again to make sure, Vec's special case using `TrustedLen` trusts the length for the reallocation but it also behaves correctly on a possible panic in `Iterator::next()`.